### PR TITLE
Fix FoulComitted + Card for Wyscout v2

### PR DESCRIPTION
**Pull request by probberechts**
- No FoulComitted event was recorded when a card was given. To be consistent with the other data providers both a FoulCommittedEvent and CardEvent are recorded now.
- Adds CardQualifiers to the FoulCommittedEvent.

### CODE REVIEW

File: kloppy/infra/serializers/event/wyscout/deserializer_v2.py [hunk 1]
```diff
@@ -537,19 +564,29 @@ def deserialize(self, inputs: WyscoutInputs) -> EventDataset:
...
+                            EventType.PASS,
+                            EventType.CLEARANCE,
+                        ]:
+                            # insert an interception event before interception passes
+                            new_events.insert(i, interception_event)
+
+                for new_event in new_events:
+                    if self.should_include_event(new_event):
+                        events.append(transformer.transform_event(new_event))
+                new_events = []
```
Comment (JanVanHaaren) [inline kloppy/infra/serializers/event/wyscout/deserializer_v2.py:589]: I'm probably overlooking something obvious, but it's not clear to me what the purpose of this line of code is.

File: kloppy/tests/test_wyscout.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def test_correct_v2_deserialization(self, event_v2_data: Path):
             == DuelType.SLIDING_TACKLE
         )
         assert (
```
Comment (JanVanHaaren) [inline kloppy/tests/test_wyscout.py:88]: Would it be possible to add a test for a `FoulCommittedEvent` followed by a `CardEvent`?

File: kloppy/infra/serializers/event/wyscout/deserializer_v2.py (hunk 1 above)
Comment (probberechts) [reply kloppy/infra/serializers/event/wyscout/deserializer_v2.py]: The `new_events = []` line has no purpose. I overlooked it. 😃

**Comment by probberechts** [comment]
I made a few more changes:
- Refactored the Wyscout tests such that you have one unit test per action type
- Use IDs instead of indexes to select events in the tests, which is much easier to maintain and to find the relevant data record in the test data
- Removed a few duplicate IDs in the Wyscout v3 test data
- Fixed the scope of the `base_dir` fixture. There is no need to recreate a constant for each function, so I've changed the scope to "session".
- Added a test for FoulCommited + Card events
- Added a prefix to the ID of synthetic events to avoid having multiple events with the same ID
- Changed the type of event IDs from int to str. This is consistent with the defined type of event IDs.

File: kloppy/infra/serializers/event/wyscout/deserializer_v2.py [hunk 3]
```diff
@@ -500,56 +526,86 @@ def deserialize(self, inputs: WyscoutInputs) -> EventDataset:
...
                         result=None,
                         qualifiers=qualifiers,
                         **generic_event_args,
                     )
+                    new_events.append(generic_event)
 
-                # Since Interception is not an event in wyscout v2 but a tag for pass, touch and duel. Therefore,
-                # we convert those duels and touch events to an interception. And insert interception before passes.
+                # Wyscout v2 does not have a seperate event type for
+                # interceptions. Interceptions are recored by adding a tag to
```
Comment (JanVanHaaren) [inline kloppy/infra/serializers/event/wyscout/deserializer_v2.py:560]: Minor: "recorded".

File: kloppy/infra/serializers/event/wyscout/deserializer_v2.py [hunk 4]
```diff
@@ -500,56 +526,86 @@ def deserialize(self, inputs: WyscoutInputs) -> EventDataset:
...
+                    generic_event = self.event_factory.build_generic(
                         result=None,
                         qualifiers=qualifiers,
                         **generic_event_args,
                     )
+                    new_events.append(generic_event)
 
-                # Since Interception is not an event in wyscout v2 but a tag for pass, touch and duel. Therefore,
-                # we convert those duels and touch events to an interception. And insert interception before passes.
+                # Wyscout v2 does not have a seperate event type for
```
Comment (JanVanHaaren) [inline kloppy/infra/serializers/event/wyscout/deserializer_v2.py:559]: Minor: "separate".

File: kloppy/tests/test_wyscout.py [hunk 5]
```diff
@@ -3,93 +3,186 @@
...
+            data_version="V2",
         )
-        assert dataset.records[2].coordinates == Point(36.0, 78.0)
+        assert dataset.dataset_type == DatasetType.EVENT
+        return dataset
+
+    def test_miscontrol_event(self, dataset: EventDataset):
+        miscontrol_event = dataset.get_event_by_id("190078351")
+        assert miscontrol_event.event_type == EventType.MISCONTROL
+        assert dataset.events[11].event_type == EventType.MISCONTROL
```
Comment (JanVanHaaren) [inline kloppy/tests/test_wyscout.py:58]: Is this `assert` statement still needed?

**Review by JanVanHaaren — COMMENTED** [review]
The changes look good to me. I left a few minor comments.

**Review by JanVanHaaren — APPROVED** [review]
Thanks!

**Comment by koenvo** [comment]
Great work @probberechts! Can you merge current master in and resolve conflicts?

**Review by koenvo — CHANGES_REQUESTED** [review]
Looking good. Please resolve merge conflicts

**Pull request merged**